apps/rig_inbound_gateway/test/event_buffer_test.exs

```diff
@@ -0,0 +1,133 @@
+defmodule EventBufferTest do
+  use ExUnit.Case, async: true
+
+  alias RigInboundGatewayWeb.EventBuffer
+
+  test "New EventBuffer starts with no events" do
+    buffer = EventBuffer.new(5)
+    assert buffer.events === []
```

Comment:
your tests rely on internals of `EventBuffer`. Also you write numbers into the buffer; either rename it to `Buffer` or write events into it. I'd prefer the latter - consider the following API:

```elixir
defp new_event(id), do: CloudEvent.parse("""
  {"specversion": "0.2", "id": "#{id}", "type": "test-event"}
""")

test "Adding events leaves the capacity unchanged and overwrites old events as soon as the buffer is filled up to its capacity.""" do
# We create a new buffer that holds at most 2 events:
buf = EventBuffer.new(2)
assert EventBuffer.capacity(buf) == 2
assert EventBuffer.all_events() == []

# We add a first event:
first_event = new_event("first")
buf = buf |> EventBuffer.add_event(first_event)
# The event is in the buffer:
assert EventBuffer.all_events() == [first_event]
# There is no more recent event yet:
assert EventBuffer.events_since("first") == []

# We add a second event to the buffer:
second_event = new_event("second")
buf = buf |> EventBuffer.add_event(second_event)
# Both events are now in the buffer:
assert EventBuffer.all_events() == [first_event, second_event]
# "second" is now more recent than "first", but there is no event newer than "second":
assert EventBuffer.events_since("first") == [second_event]
assert EventBuffer.events_since("second") == []

# Adding a third event should remove the first one:
third_event = new_event("third")
buf = buf |> EventBuffer.add_event(third_event)
# The first event is no longer in the buffer:
assert EventBuffer.all_events() == [second_event, third_event]
# "third" is now more recent than "second", but there is no event newer than "third":
assert EventBuffer.events_since("second") == [third_event]
assert EventBuffer.events_since("third") == []
# Since "first" is no longer in the buffer, _all_ events are newer than "first":
assert EventBuffer.events_since("first") == [second_event, third_event]

# The capacity hasn't changed during the modifications:
assert EventBuffer.capacity(buf) == 5
end
```